### ISSUE TITLE
Application does not load completely sometimes.

### ISSUE THREAD

**Issue by IamHDT** [freeCodeCamp/freeCodeCamp#15972]
This issue is already happening on https://beta.freecodecamp.org/en/challenges/join-the-freecodecamp-community/learn-how-freecodecamp-works
#### Screenshot
<img width="1540" alt="screen shot 2017-10-18 at 21 51 14" src="https://user-images.githubusercontent.com/26920327/31725478-b16f4b0c-b44e-11e7-9521-eaabb1e8689b.png">

**Comment by raisedadead** [comment]
I can reproduce the timeout on the bundle download by turning on the the throttling in Chrome Dev Tools.

**Comment by anhtran1906** [comment]
I can't reproduce this issue on my local machine. Could you tell me how to reproduce it?
Also I was involved in the "No Super Blocks" bug and want to understand why #16047 relates to this issue. Could someone explain for me to see if I could continue helping with this issue?

**Comment by raisedadead** [comment]
I can reproduce on the beta site by throttling at Slow/Fast 3G. The AJAX request for the bundle timesout.

@BerkeleyTrue 
I am not sure, if we can leverage any of the webpack's code spliting? or some form of a pre-loading animation, etc. to wrap around the intial bundle loading.

Related report: https://github.com/freeCodeCamp/freeCodeCamp/issues/16129

**Comment by raisedadead** [comment]
Nope the network speed is not the only issue here, I tried disabling the cache and it did work without issues on the slow network.

**Comment by BerkeleyTrue** [comment]
I'm guessing something being saved to localStorage is not the correct format?

**Comment by raisedadead** [comment]
@BerkeleyTrue I dont have clean analysis, yet. But the consistent conditions to reproduce this are:

On production:
- Toggle the network throttle to a slow connection (available in Chrome).
-  Do NOT disable the cache (Available in Chrome).
- While the intial challenge page is loading and the map looks dash-ed, click on the UI.
- Note the console for the errors.

On Local:
 No analysis yet.

To me, it seems that the initial action while the bundle has not completely loaded, triggers methods which are defined but not yet ready.

This throws of the errors.

Also, wierdly it cancels off the further bundle loading.

**Comment by BerkeleyTrue** [comment]
@raisedadead I don't think it causing the bundle to stop loading. I believe that it is causing an uncaught error to collapse the rx pipeline. 

The map ui initially loads just the superblocks (SSR). The only challenge data loaded other then this is current challenge. I'm suspecting the action of clicking on a superblock in the map UI before the `fetchChallenges.complete` will cause an uncaught exception. 

Try disabling the fetchChallengesEpic and clicking on a map ui. This may be the cause.

**Comment by raisedadead** [comment]
Sure, will check first thing tomorrow.

**Comment by QuincyLarson** [comment]
@raisedadead Were you able to look into this? I was QA'ing beta yesterday on a slow connection and I attempted to click a superblock before it loaded, and it crashed the app ("oops" toasts everywhere). 

We need to speed up the load of the map, but in the meantime, we should put a spinner there until it finishes loading.

**Comment by BerkeleyTrue** [comment]
> we should put a spinner there until it finishes loading.

This is unnecessary and a regression. We just need to find what action is crashing due to unexpected state.

**Comment by QuincyLarson** [comment]
@BerkeleyTrue I agree we need to find what action is crashing due to unexpected state. I suggested the spinner because we have a lot of competing priorities, and may be able to paper over the issue in the meantime. If you can fix the underlying problem in time for launch, by all means.

**Comment by momitrahman** [comment]
\# I am facing this same issue for long time. Application can't load properly and then 'Something went wrong' message popup. 

\# In browser console in both Firefox and Chrome browser this message logged:
```
Error: XMLHttpRequest timeout Stack trace: a/h<@https://beta.freecodecamp.org/js/bundle-fb8910782c4be3178828.js:66:8995
```
\# I want to clear some fact  when I face this loading problem:
- Last month I was using 8Mbps speed then application was load properly.
- This month i change my speed 8Mbps to 4Mbps then the problem started.
- I tried after clear my cache and in some other browser same thing happen.
- My friend and i using same ISP internet so i checked on his PC with his 8Mbps speed connection but there was no loading problem.
- Then i tried load application in different speed using Chrome developer tools network speed control options then its can't load in when speed less than 6 Mbps but 6Mbps <  8 Mbps  can load sometimes. 
- I also tried on different ISP connection in some of my friends pc then same things happen in slow speed connection.
- There is last thing i want to mention that i find a very common pattern in all pc  by using Chrome browser dev tools that Application load properly when loading 'ms' is around 8000ms when more than 8000ms the problem occurs. 

N.B : If my English gone bad and my comment cant help you (@raisedadead) out then I am extremely sorry for that.
